sklearn/linear_model/_glm/glm.py

```diff
@@ -268,7 +264,29 @@ def fit(self, X, y, sample_weight=None):
             y_numeric=True,
             multi_output=False,
         )
-
+        if hasattr(self.alpha, "__iter__") and not isinstance(self.alpha, str):
+            for i, val in enumerate(self.alpha):
+                check_scalar(
+                    val,
+                    name=f"alpha at index {i}",
+                    target_type=numbers.Real,
+                    min_val=0.0,
+                    include_boundaries="left",
+                )
+            self.alpha = np.asarray(self.alpha, dtype=np.float_).ravel()
```

Comment:
I will change to np.float64, I don't like the idea of using X.dtype in the off chance that X.dtype is integer dtype. Even if X is all integers it would be very wrong for the penalty terms to be integers